apps/mull-ui/src/app/components/event-page-info/event-page-info.tsx

```diff
@@ -77,9 +88,31 @@ export const EventPageInfo = ({
         <p className="row-text">{event.participants?.map((p) => p.name).join(', ')}</p>
         <FontAwesomeIcon icon={faUserPlus} className="event-page-icon color-green" />
       </div>
-      <MullButton className="event-page-button" onClick={handleMullButton} type={buttonType}>
-        {isReview ? 'Create' : 'Join'}
-      </MullButton>
+      {joined ? (
+        <MullButton
+          className="event-page-button event-page-joined-button"
+          onClick={(e) => {
+            e.stopPropagation();
+            setJoined(!joined);
+            removeJoinedEvent({ variables: { eventId, userId } });
+          }}
+          type={buttonType}
+        >
+          {'Leave'}
+        </MullButton>
+      ) : (
+        <MullButton
+          className="event-page-button"
+          onClick={(e) => {
+            e.stopPropagation();
+            setJoined(!joined);
+            joinEvent({ variables: { eventId, userId } });
+          }}
+          type={buttonType}
+        >
+          {isReview ? 'Create' : 'Join'}
+        </MullButton>
+      )}
```

Comment:
Most of the code is being duplicated. You can simplify this by only putting a conditional on the differences. for example:

```tsx
      <MullButton
        className={`event-page-button ${joined ? 'event-page-joined-button' : ''}`}
        onClick={(e) => {
          e.stopPropagation();
          setJoined(!joined);
          if (joined) {
            removeJoinedEvent({ variables: { eventId, userId } });
          } else {
            joinEvent({ variables: { eventId, userId } });
          }
        }}
        type={buttonType}
      >
        {joined ? 'Leave' : isReview ? 'Create' : 'Join'}
      </MullButton>
```

My solution might have some problems because I didn't test it, but you get the point